### ISSUE TITLE
404 Error during WebSocket handshake

### ISSUE THREAD

**Issue by johnthad** [realityforge/gwt-websockets-example#1]
Hello. I have tried running this example in both devmode (well, with GWT 2.7.0, it's SDM) and as WAR deployed in Tomcat 8. Either way I get a 404 error when I try to connect (below is the Chrome console when running from Tomcat). Do you have an idea what is wrong or what to look for?

```
example-0.js:695 WebSocket connection to 'ws://10.1.2.2:8080/gwt-websockets-example/chat' failed: Error during WebSocket handshake: Unexpected response code: 404
example-0.js:717 Tue Oct 06 16:54:30 GMT-400 2015 org.realityforge.gwt.websockets.example.client.Example
WARNING: WebSocket @ Error
ReadyState:CLOSED
URL:ws://10.1.2.2:8080/gwt-websockets-example/chat
BinaryType:ARRAYBUFFER
BufferedAmount:0
Extensions:
Protocol:
example-0.js:717 Tue Oct 06 16:54:30 GMT-400 2015 org.realityforge.gwt.websockets.example.client.Example
WARNING: WebSocket @ Close
ReadyState:CLOSED
URL:ws://10.1.2.2:8080/gwt-websockets-example/chat
BinaryType:ARRAYBUFFER
BufferedAmount:0
Extensions:
Protocol:
```

**Comment by realityforge** [comment]
Hi, 

The error indicates that the client can not find the server component of this. In particular it seems that the [ChatServer](https://github.com/realityforge/gwt-websockets-example/blob/master/src/main/java/org/realityforge/gwt/websockets/example/server/ChatServer.java) is not deployed to the url ws://10.1.2.2:8080/gwt-websockets-example/chat

So I would check - does Tomcat 8 support javax.websocket? Is the compiled class in the deployed war?

**Comment by johnthad** [comment]
Yes, Tomcat 8 supports javax.websocket. It [comes with several examples](http://tomcat.apache.org/tomcat-8.0-doc/web-socket-howto.html), and I can run those on my installation. And the ChatServer.class is in the correct place -- gwt-websockets-example/WEB-INF/classes/org/realityforge/gwt/websockets/example/server. I'm also getting the same error with localhost:8080 (vs 10.1.2.2:8080).

**Comment by johnthad** [comment]
Sorry to leave you hanging, but I just got back to this today.

I figured it out--the WAR I was building was improperly including several JARs--like `javaee-api-7.0.jar` and `jsr305-2.0.1.jar`--in `WEB-INF/lib`. When I removed all but `gwt-websockets-1.1.2.jar` and `gwt-servlet-2.7.0.jar`, the example runs when deployed on Tomcat 8. 

As for running in GWT devmode, that ain't gonna happen because devmode runs Jetty 6.1. If I'm to test a websocket in development, I'm gonna have to run Tomcat 8 or Jetty 9 as external server.

Thanks for the library. I expect to put it to good use in my next project.
